wfl/calculators/espresso.py

```diff
@@ -59,57 +51,11 @@ class Espresso(WFLFileIOCalculator, ASE_Espresso):
-                 calculator_exec=None, **kwargs):
+                 workdir=None, scratchdir=None, **kwargs):
 
         kwargs_command = deepcopy(kwargs)
 
-        # check for various Espresso versions
-        # NOTE: should we be doing this much massaging of inputs, or should we make the user keep up
-        # with their ASE Espresso version?
-        if EspressoProfile is not None:
-            # new version, command and ASE_ESPRESSO_COMMAND deprecated
-            if "command" in kwargs_command:
-                raise ValueError("Espresso calculator defines EspressoProfile, but deprecated 'command' arg was passed")
-
-            if calculator_exec is not None:
-                # check for conflicts, wrong format
-                if "profile" in kwargs_command:
-                    raise ValueError("Cannot specify both calculator_exec and profile")
-                if " -in " in calculator_exec:
-                    raise ValueError("calculator_exec should not include espresso command line arguments such as ' -in PREFIX.pwi'")
-
-                # newer syntax, but pass binary without a keyword (which changed from "argv" to "exc"
-                # to "binary" over time), assuming it's first argument
-                argv = shlex.split(calculator_exec)
-                try:
-                    kwargs_command["profile"] = EspressoProfile(argv=argv)
-                except TypeError:
-                    binary, parallel_info = parse_genericfileio_profile_argv(argv)
-                    # argument names keep changing (e.g. pseudo_path -> pseudo_dir), just pass first two as positional
-                    # and hope order doesn't change
-                    if "pseudo_dir" not in kwargs_command:
-                        raise ValueError(f"calculator_exec also requires pseudo_dir to create EspressoProfile")
-                    kwargs_command["profile"] = EspressoProfile(binary, kwargs_command.pop("pseudo_dir"),
-                                                                parallel_info=parallel_info)
-            elif "profile" not in kwargs_command:
-                raise ValueError("EspressoProfile is defined but neither calculator_exec nor profile was specified")
-
-            # better be defined by now
-            assert "profile" in kwargs_command
-        else:
-            # old (pre EspressoProfile) version
-            if "profile" in kwargs_command:
-                raise ValueError("EspressoProfile is not defined (old version) but profile was passed")
-
-            if calculator_exec is not None:
-                if "command" in kwargs_command:
-                    raise ValueError("Cannot specify both command and calc_exec")
-
-                kwargs_command["command"] = f"{calculator_exec} -in PREFIX.pwi > PREFIX.pwo"
-
-            # command or env var must be set
-            assert "command" in kwargs_command or "ASE_ESPRESSO_CALCULATOR" in os.environ
+        assert "profile" in kwargs_command
```

Comment:
Please don't require the new version.  Just let the user pass whatever arguments they want, and if it's compatible with their ASE version, it'll work, if not, not.

Reply:
Removed, see #325